### PR TITLE
Add support for LoongArch

### PR DESCRIPTION
**Pull request by wenghongquan**
Add support for LoongArch, please review it, thanks!

### CODE REVIEW

**Comment by ringabout** [comment]
@wenghongquan  Hi, do you use/test this arch with Nim?

File: compiler/installer.ini [hunk 1]
```diff
@@ -6,7 +6,7 @@ Name: "Nim"
 Version: "$version"
 Platforms: """
   windows: i386;amd64
-  linux: i386;hppa;ia64;alpha;amd64;powerpc64;arm;sparc;sparc64;m68k;mips;mipsel;mips64;mips64el;powerpc;powerpc64el;arm64;riscv32;riscv64
+  linux: i386;hppa;ia64;alpha;amd64;powerpc64;arm;sparc;sparc64;m68k;mips;mipsel;mips64;mips64el;powerpc;powerpc64el;arm64;riscv32;riscv64;loongarch64
```
Comment (dom96) [inline compiler/installer.ini]: Not familiar with this architecture, but perhaps we shouldn't be generating official C sources for it (if it's very rare). But I dunno, maybe C source bloat isn't too big of a concern right now.

**Review by dom96 — APPROVED** [review]
Other than my comment seems good.

File: compiler/installer.ini (hunk 1 above)
Comment (ghost) [reply compiler/installer.ini]: @dom96 yeah, I think you're right, Araq did remove e2k from installer.ini in my PR so I'm assuming it's to not make the csources too big.

Comment (Araq) [reply compiler/installer.ini]:
```suggestion
  linux: i386;hppa;ia64;alpha;amd64;powerpc64;arm;sparc;sparc64;m68k;mips;mipsel;mips64;mips64el;powerpc;powerpc64el;arm64;riscv32;riscv64
```

**Comment by ringabout** [comment]
> @wenghongquan Hi, do you use/test this arch with Nim?

Out of curiosity, afaik the chips with this arch are hard/expensive to purchase on the markets in China. Do you have a PC with LoongArch or for other reasons you need to use this arch?

请问你使用的是龙芯嵌入式电脑还是 PC?

**Comment by wenghongquan** [comment]
> > @wenghongquan Hi, do you use/test this arch with Nim?
> 
> Out of curiosity, afaik the chips with this arch are hard/expensive to purchase on the markets in China. Do you have a PC with LoongArch or for other reasons you need to use this arch?
> 
> 请问你使用的是龙芯嵌入式电脑还是 PC?

I'm using PC with LoongArch now.

**Pull request merged**